test/unit/person_test.rb

```diff
@@ -1407,4 +1408,327 @@ def test_updated_not_changed_when_adding_notifiee_info
     assert_equal [project, project2, project3], person.projects.sort_by(&:id)
     assert_equal [project,project2], person.administered_projects.sort_by(&:id)
   end
+
+  test 'merge simple attributes' do
+    person_to_keep = FactoryBot.create(:min_person)
+    other_person = FactoryBot.create(:max_person)
+    other_person_attributes = {}
+    simple_attributes.each do |attribute|
+      other_person_attributes[attribute] = other_person.send(attribute)
+    end
+
+    disable_authorization_checks { person_to_keep.merge(other_person) }
+    person_to_keep.reload
+
+    assert_equal 'Minimal', person_to_keep.last_name
+    assert_equal 'minimal_person@email.com', person_to_keep.email
+    attributes = simple_attributes - %i[last_name email]
+    attributes.each do |attribute|
+      assert_equal other_person_attributes[attribute], person_to_keep.send(attribute),
+                   "Should copy #{attribute} if empty in person to keep"
+    end
+  end
+
+  test 'merge annotations' do
+    person_to_keep = FactoryBot.create(:min_person)
+    person_to_keep.annotate_with(%w[golf merging], 'expertise', person_to_keep)
+    person_to_keep.annotate_with(%w[merger], 'tool', person_to_keep)
+    person_to_keep.save!
+    person_to_keep.reload
+    orig_annotations = {}
+    annotation_types.each do |annotation_type|
+      orig_annotations[annotation_type] = person_to_keep.send(annotation_type)
+    end
+    other_person = FactoryBot.create(:max_person)
+    other_annotations = {}
+    annotation_types.each do |annotation_type|
+      other_annotations[annotation_type] = other_person.send(annotation_type)
+    end
+
+    disable_authorization_checks { person_to_keep.merge(other_person) }
+    person_to_keep.reload
+
+    annotation_types.each do |annotation_type|
+      assert_equal (orig_annotations[annotation_type] + other_annotations[annotation_type]).compact.uniq.sort,
+                   person_to_keep.send(annotation_type).sort,
+                   "Should copy #{annotation_type} if not present in person to keep"
+    end
```

Comment:
would be good to have an explicit test that checks the values for tools and expertise

Reply:
The test already copies the values and checks for them... or would you prefer something  hard-coded in the test? as in:
```
assert_equal ["fishing", "golf", "merging"].sort, person_to_keep.expertise.sort
```
The above is effectively the test that is taking place, but if a new annotation is added to the list, it is tested automatically... but I can change it, if you prefer...

Reply:
yes it would good to add that. It would be clearer, and will also avoid a mistake in the test allowing it to pass unexpectedly.
Also worth remembering that unit tests also act as documentation about what the method is meant to be doing.

Reply:
Resolved in [f19cc32](https://github.com/seek4science/seek/pull/1683/commits/f19cc32e6b6b33ffc7a47cd8c101a5e18e967d64).